src/content/extensions/multipleroutes/src/options.js

```diff
@@ -1,97 +1,114 @@
-      'value': nonProxiedUdp
+  if (pn.webRTCIPHandlingPolicy !== undefined) {
+    pn.webRTCIPHandlingPolicy.set({
+      value: mapRadioIdToPolicy[i]
+    });
+  } else {
+    var mapping = mapRadioIdToPolicy[i];
+    pn.webRTCMultipleRoutesEnabled.set({
+      value: mapping[0]
+    }, function() {
+      if (pn.webRTCNonProxiedUdpEnabled !== undefined) {
+        pn.webRTCNonProxiedUdpEnabled.set({
+          value: mapping[1]
+        });
+      }
     });
-  } catch (err) {
-    console.log('setting webRTCNonProxiedUdpEnabled is not supported.');
   }
 }
 
-function restoreRadios(multiRoutes, nonProxiedUdp) {
-  var radios = document.getElementsByName('routeselection');
-  if (multiRoutes) {
-    if (nonProxiedUdp) {
-      radios[0].checked = true;
-    } else {
-      radios[0].checked = false;
-      alert(
-          'The extension does not support the configuration where multiple ' +
-              'routes is enabled but non-proxied udp is disabled.'
-      );
-    }
-  } else {
-    if (nonProxiedUdp) {
-      radios[1].checked = true;
-    } else {
-      radios[2].checked = true;
-    }
-  }
+function restoreRadios(policy) {
+  var radios = document.getElementsByName('ip_policy_selection');
+  radios[mapPolicyToRadioId[policy]].checked = true;
 }
 
-// Restores checkbox states.
-function restoreMultiRoutesOption() {
-  var multiRoutes = true;
-  chrome.privacy.network.webRTCMultipleRoutesEnabled.get({},
-    function(details) {
-      multiRoutes = details.value;
-      restoreRadios(multiRoutes, true);
-      restoreNonProxiedUdpOption(multiRoutes);
+function restoreOption() {
+  if (pn.webRTCIPHandlingPolicy !== undefined) {
+    pn.webRTCIPHandlingPolicy.get({}, function(details) {
+      restoreRadios(details.value);
     });
+  } else {
+    getPolicyFromBooleans(false, function(policy) {
+      restoreRadios(policy);
+    });
+  }
 }
 
-function restoreNonProxiedUdpOption(multiRoutes) {
-  try {
-    var nonProxiedUdp = true;
-    chrome.privacy.network.webRTCNonProxiedUdpEnabled.get({},
-      function(details) {
-        nonProxiedUdp = details.value;
-        restoreRadios(multiRoutes, nonProxiedUdp);
-        document.getElementById('for_multirouteOffUdpOff_notSupported').
-          innerHTML = '';
-      });
-  } catch (err) {
-    console.log(err);
-    document.getElementById('multirouteOffUdpOff').disabled = true;
-    document.getElementById('multirouteOffUdpOff_Section').style.color = 'gray';
-    var chromeVersion = /Chrome\/([0-9.]+)/.exec(navigator.userAgent)[1];
-    document.getElementById('for_multirouteOffUdpOff_notSupported').innerHTML +=
-      ' Current version is: ' + chromeVersion;
+// Returns the supported modes in the UI option.
+function getSupportedModes() {
+  if (pn.webRTCNonProxiedUdpEnabled === undefined) {
+    return [true, false, true, false];
   }
+  if (pn.webRTCIPHandlingPolicy === undefined) {
+    return [true, false, true, true];
+  }
+  return [true, true, true, true];
```

Comment:
Maybes it's Mode1, Mode2, Mode3, Mode4, etc, to match the element ID.  That might make the mode hiding stuff below easier.